changes.go

```diff
@@ -0,0 +1,103 @@
+package automerge
+
+// #include "automerge.h"
+import "C"
+import (
+	"encoding/hex"
+	"runtime"
+	"time"
+)
+
+// ChangeHash is a 32-byte SHA-256 hash identifying an automerge change.
+// Like a git commit, the hash encompasses both the changes made,
+// any metadata (like commit message, or timestamp)
+// and any changes on which this change depends.
+type ChangeHash []byte
```

Comment:
I had originally gone with []byte because you can more easily represent nil; that said, it doesn't seem like we need to represent a nil change hash. Updated!